src/magic.c

```diff
@@ -91,6 +92,13 @@ TbBool can_cast_spell_f(PlayerNumber plyr_idx, PowerKind pwkind, MapSubtlCoord s
     if (!is_power_available(plyr_idx, pwkind)) {
         return false;
     }
+    struct PlayerInfo* player = get_player(plyr_idx);
+    if (player->work_state == PSt_FreeCtrlDirect)
+    {
+        return true;
+    }
+    else
+    {
     TbBool cast_at_xy;
```

Comment:
Could you indent the entire else statement, to the level of "return true" above. So 4 spaces in front lines 102-167.